src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,119 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int ELEMENT_NOT_FOUND = -1;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        provideCapacity(size + 1);
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkAddID(index);
```

Comment:
Here you may safely use same checkIndex method as everywhere else as case when (index == size) will not get here due to if above
```suggestion
        checkIndex(index);
```